protocol/storage/miner_test.go

```diff
@@ -297,18 +304,319 @@ func TestOnCommitmentAddedToChain(t *testing.T) {
+
+		handlers := successMessageHandlers(t)
+		handlers["submitPoSt"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+			t.Error("Should not have called submit post")
+			return [][]byte{}, nil
+		}
+		api.messageHandlers = handlers
+
+		height := uint64(10500)
+		api.blockHeight = types.NewBlockHeight(height)
+		block := &types.Block{Height: types.Uint64(height)}
+		ts, err := types.NewTipSet(block)
+		require.NoError(t, err)
+
+		miner.OnNewHeaviestTipSet(ts)
+
+		// OnNewHeaviestTipSet spawns a goroutine that should exit immediately dependencies are faked out
+		time.Sleep(1 * time.Second)
+
+		assert.Equal(t, 0, len(logger.ErrorMessages))
+		assert.Equal(t, 0, len(logger.InfoMessages))
+	})
+
+	t.Run("Logs error if past proving period", func(t *testing.T) {
+		// create new miner with deal in the accepted state and mapped to a sector
+		api, miner, _, logger := minerWithAcceptedDealTestSetup(t, proposalCid, sector.SectorID)
+
+		handlers := successMessageHandlers(t)
+		handlers["submitPoSt"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+			t.Error("Should not have called submit post")
+			return [][]byte{}, nil
+		}
+		api.messageHandlers = handlers
+
+		height := uint64(50500)
+		api.blockHeight = types.NewBlockHeight(height)
+		block := &types.Block{Height: types.Uint64(height)}
+		ts, err := types.NewTipSet(block)
+		require.NoError(t, err)
+
+		miner.OnNewHeaviestTipSet(ts)
+
+		// OnNewHeaviestTipSet spawns a goroutine that should exit immediately dependencies are faked out
+		time.Sleep(1 * time.Second)
+
+		assert.Equal(t, 1, len(logger.ErrorMessages))
+		assert.Contains(t, logger.ErrorMessages[0], "too late start")
+	})
+}
+
+func successMessageHandlers(t *testing.T) messageHandlerMap {
+	handlers := messageHandlerMap{}
+	handlers["isBootstrapMiner"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+		return mustEncodeResults(t, false), nil
+	}
+
+	handlers["getSectorCommitments"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+		commitments := map[string]types.Commitments{}
+		commitments["42"] = types.Commitments{}
+		return mustEncodeResults(t, commitments), nil
+	}
+	handlers["getProvingPeriod"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+		return mustEncodeResults(t, types.NewBlockHeight(20003), types.NewBlockHeight(40003)), nil
+	}
+	handlers["submitPoSt"] = func(a address.Address, v types.AttoFIL, p ...interface{}) ([][]byte, error) {
+		return [][]byte{}, nil
+	}
+	return handlers
+}
+
+type testNode struct{}
+
+func (tn *testNode) BlockService() bserv.BlockService           { return nil }
+func (tn *testNode) Host() host.Host                            { return nil }
+func (tn *testNode) SectorBuilder() sectorbuilder.SectorBuilder { return &testSectorBuilder{} }
+
+type testSectorBuilder struct{}
```

Comment:
OBSERVATION: This is a lot of test code, which is unfortunate because only two of the methods are consumed by the storage miner (`AddPiece` and `ReadPieceFromSealedSector`). 

I suggest that you modify the `storage.node` interface such that

```
type node interface {
	BlockService() bserv.BlockService
	Host() host.Host
	SectorBuilder() sectorbuilder.SectorBuilder
}
```

becomes

```
type node interface {
	BlockService() bserv.BlockService
	Host() host.Host
	SectorBuilder() someNewInterfaceWithJustWhatStorageMinerNeeds
}

type someNewInterfaceWithJustWhatStorageMinerNeeds interface {
    AddPiece(...)
    ReadPieceFromSealedSector(...)
}
```

You'd need to do the same thing in the retrieval miner code.

Reply:
This is a good idea, but out of scope for this PR.